security/tokens_and_compartments.rst

```diff
@@ -0,0 +1,199 @@
+A CompartmentDefinition first and foremost defines abstract relationships between resources, but it becomes useful once you combine it with a way of specifying a concrete focus resource instance.
+In SMART on FHIR, the launch context can do that. As per the SMART on FHIR `Scopes and Launch Context <https://hl7.org/fhir/smart-app-launch/scopes-and-launch-context.html>`_, applications can request information about the context from which they were launched, e.g. information about the currently active EHR session. 
+Some of the launch contexts mentioned in SMART on FHIR map to CompartmentDefinitions. For example, the launch context 'launch/patient' and 'launch/encounter' map to the compartment 'Patient' and 'Encounter'. Please note that launch contexts can be extended for any resource type, but not all resource types have a matching CompartmentDefinition, e.g. 'Location'.
+It is up to the authorization and the EHR to establish and exchange the launch context. See :ref:`firely_auth_endpoints_launchcontext` on how this can be achieved in Firely Auth.
+
+Launch contexts combined with a CompartmentDefinition defines a -- what we call -- compartment in Firely Server. Firely Server will restrict the access of a client to a compartment if:
+
+* the corresponding CompartmentDefinition is known to Firely Server, see :ref:`conformance` for options to provide them.
+* the OAuth2 Token contains a claim with the same name as the CompartmentDefinition.code (but it must be lowercase).
+
+.. _accesstokens:
+
+Tokens requirements
+-------------------
+
+Note that it is out-of-scope of the SMART on FHIR specification to define the format of the access token.
+SMART on FHIR solely defines the scope and launch context syntax. From a client perspective an access token is fully opaque.
+
+Firely Server is not bound to any specific authorization server, as long as the access token (JSON Web Token (JWT) or a Reference Token) contains a set of minimal information:
+
+* the ``iss`` claim with the base url of the OAuth server
+* the ``aud`` the same value you've entered in ``SmartAuthorizationOptions.Audience``
+* the ``scope`` field with the scopes granted by this access token
+* optionally, the compartment claim, if you'd like to limit this token to a certain compartment. For example, in case of Patient data access where the ``launch/patient`` scope is used, include the ``patient`` claim. Such a claim should be included by the authorization server if requested by using a context scope or launching a part of an EHR launch. 
+
+.. warning:: Firely Server will not enforce any access control for resources outside of the specified compartment. Some compartment definitions do not include crucial resource types like 'Patient' or their corresponding resource type, i.e. all resources of this type regardless of any claims in the access token will be returned if requested. Please use compartments other than Patient with caution! Additional custom access control is highly recommended.
```

Comment:
I would already mention here that you should limit access to the resource types outside of the compartment using the scopes.